go/vt/vttablet/tabletmanager/healthcheck.go

```diff
@@ -310,11 +310,12 @@ func (agent *ActionAgent) runHealthCheckLocked() {
 			if err != nil {
 				log.Infof("Error updating mysql port in tablet record (will try again at healthcheck interval): %v", err)
 			} else {
-				// save the port so we don't update it again next time
-				// we do the health check.
+				// save the port so we don't update it
+				// again next time we do the health check.
 				agent.mutex.Lock()
 				topoproto.SetMysqlPort(agent._tablet, mysqlPort)
 				agent._waitingForMysql = false
+				agent.skipMysqlPortCheck = true
```

Comment:
I'm wondering if we can make this more readable. I would suggest the following:

- Move this if block into its own method e.g. "getMysqlPortLocked"? This way we can return early if we don't need to get the port.
- Introduce a new field instead of reusing the existing one. I'm concerned that we're turning the existing field from an immutable into a mutable one and mix two different things into one. How about renaming the existing one to "skipMysqlPortCheckForTesting" to stress that it's only relevant for vtcombo which is basically testing code. Then we can have another field which expresses that we already did the check once.
- Is there a way to test this? e.g. run the healthcheck function twice and verify that we updated the topology only once?